src/protocol/mod.rs

```diff
@@ -21,7 +22,10 @@ use std::hash::Hash;
 /// See `IPAProtocolStep` for a canonical implementation of this trait. Every time we switch to
 /// use a new circuit, there will be an additional struct/enum that implements `Step`, but eventually
 /// it should converge to a single implementation.
-pub trait Step: Copy + Clone + Debug + Eq + Hash + Send + 'static {}
+pub trait Step:
+    Copy + Clone + Debug + Eq + Hash + Send + TryFrom<String> + ToString + 'static
```

Comment:
the official guidance is to implement `Display` instead: https://doc.rust-lang.org/std/string/trait.ToString.html

Reply:
fixed